File: app/src/main/java/com/ezen/lolketing/adapter/LeagueInfoAdapter.java
package com.ezen.lolketing.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.ezen.lolketing.fragment.OutlineFragment;
import com.ezen.lolketing.fragment.PrizeFragment;
import com.ezen.lolketing.fragment.ProgressFragment;

public class LeagueInfoAdapter extends FragmentStatePagerAdapter {
    private int leagueInfoPagerCount;

    public LeagueInfoAdapter(@NonNull FragmentManager fm, int behavior) {
        super(fm, behavior);
        this.leagueInfoPagerCount = behavior;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                OutlineFragment outlineFragment = new OutlineFragment();
                return outlineFragment;
            case 1:
                ProgressFragment progressFragment = new ProgressFragment();
                return progressFragment;
            case 2:
                PrizeFragment prizeFragment = new PrizeFragment();
                return  prizeFragment;
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return leagueInfoPagerCount;
    }
}

File: app/src/main/java/com/ezen/lolketing/adapter/LeagueInfoAdapter.kt
package com.ezen.lolketing.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class LeagueInfoAdapter(fragmentActivity: FragmentActivity): FragmentStateAdapter(fragmentActivity) {
    private var fragments : ArrayList<Fragment> = ArrayList()

    override fun getItemCount(): Int = fragments.size

    override fun createFragment(position: Int): Fragment = fragments[position]

    fun addFragment(fragment: Fragment) {
        fragments.add(fragment)
    }

}


